cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -358,11 +369,27 @@ private void deployConfigAndRefreshStaleServices(ClustersResourceApi clustersRes
                 .map(it -> it.getName() + ": " + it.getClientConfigStalenessStatus())
                 .collect(Collectors.joining(", ")));
         List<ApiCommand> commands = clustersResourceApi.listActiveCommands(stack.getName(), SUMMARY).getItems();
-        ApiCommand deployClientConfigCmd = checkedMeasure(
-                () -> getApiCommand(commands, "DeployClusterClientConfig", stack.getName(), clustersResourceApi::deployClientConfig),
-                LOGGER,
-                "The DeployClusterClientConfig command registration to CM took {} ms");
-        pollDeployConfig(deployClientConfigCmd);
+
+        BigDecimal deployClientConfigCommandId;
+        if (syncApiCommandPollerConfig.isSyncApiCommandPollingEnaabled(
+                Crn.safeFromString(stack.getResourceCrn()).getAccountId())) {
+            deployClientConfigCommandId = multiCheckedMeasure(
+                    () -> (Benchmark.MultiCheckedSupplier<BigDecimal, ApiException, CloudbreakException>)
+                            () -> clouderaManagerSyncApiCommandIdProvider.executeSyncApiCommandAndGetCommandId(
+                                    syncApiCommandPollerConfig.getDeployClusterClientConfigCommandName(), clustersResourceApi, stack,
+                                    () -> clustersResourceApi.deployClientConfig(stack.getName())),
+                    LOGGER,
+                    "The DeployClusterClientConfig command registration to CM took {} ms")
+                    .get();
```

Comment:
this is very hard to read. please refactor a bit into more readable format

Reply:
i could not really make it smaller, but i could put these into different smaller methods